src/components/Form/Combobox.js

```diff
@@ -0,0 +1,333 @@
+  onClickOutside = () => {
+    this.setState({ open: false });
+  }
+
+  getSelectHandler = itemKey => memoize(() => this.onSelect(itemKey));
+
+  getSelectedItems() {
+    const { items } = this.props;
+
+    return items.filter(item => item.selected === true);
+  }
+
+  handleClickOutside() {
+    this.setState({ open: !this.state.open });
+  }
+
+  renderInput() {
+    const {
+      id,
+      disabled,
+      required,
+      labelMultiselect,
+      onChange,
+      placeholder,
+      readOnly,
+      value,
+    } = this.props;
+
+    const formatLabel = () => {
+      if (readOnly) {
+        const selectedItems = this.getSelectedItems();
+        const count = selectedItems.length;
+
+        if (count > 1) {
+          return `${count} ${labelMultiselect}`;
+        } else if (count === 1) {
+          return selectedItems[0].label;
+        }
+      } else {
+        return value;
+      }
+
+      return '';
+    };
+
+    const label = formatLabel();
+
+    const classNames = cx(
+      'slds-combobox__input',
+      { 'slds-combobox__input-value': !!label },
+    );
+
+    const iconRight = readOnly ? 'down' : 'search';
+
+    return (
+      <InputRaw
+        aria-controls={`listbox-${id}`}
+        autoComplete="off"
+        className={classNames}
+        disabled={disabled}
+        iconRight={iconRight}
+        iconRightOnClick={this.onToggle}
+        id={`combobox-${id}`}
+        onChange={onChange}
+        onClick={this.onToggle}
+        onKeyDown={readOnly ? null : this.onInputKeyDown}
+        placeholder={placeholder}
+        readOnly={readOnly}
+        required={required}
+        role="textbox"
+        value={label}
+      />
+    );
+  }
+
+  renderComboboxItems() {
+    const { items } = this.props;
+
+    return items.map(item => (
+      <ComboboxDropdownListItem
+        icon={item.icon}
+        id={`listbox-option-${item.key}`}
+        isHeader={item.isHeader}
+        key={item.key}
+        onClick={this.getSelectHandler(item.key)}
+        selected={item.selected}
+      >
+        {item.label}
+      </ComboboxDropdownListItem>
+    ));
+  }
+
+  renderInlineListbox() {
+    const selectedItems = this.getSelectedItems();
+
+    return (
+      <Listbox className="slds-listbox_inline">
+        {selectedItems.map(({ icon, key, label }) => (
+          <Pill
+            icon={icon ? <Icon icon={icon.icon} size="small" sprite={icon.sprite} /> : null}
+            key={key}
+            label={label}
+            title={label}
+            onClose={this.getSelectHandler(key)}
+          />
+        ))}
+      </Listbox>
+    );
+  }
+
+  renderSelectedPills() {
+    const { onSelect } = this.props;
+
+    const selectedItems = this.getSelectedItems();
+
+    return (
+      <Listbox className="slds-listbox_inline slds-p-top_xxx-small">
+        {selectedItems.map(({ icon, key, label }) => (
+          <Pill
+            icon={icon ? <Icon icon={icon.icon} size="small" sprite={icon.sprite} /> : null}
```

Comment:
Could be moved to a function, see above.

Reply:
@PeterSiegl I disagree, this should not be a function